lib/handler/retry-handler.js

```diff
@@ -157,13 +157,24 @@ class RetryHandler {
       Array.isArray(statusCodes) &&
       !statusCodes.includes(statusCode)
     ) {
-      cb(err)
+      shouldRetryCb(err)
+      return
+    }
+
+    // If the error is a body timeout we want to abort the request
+    // as the server could be still sending data and we want to avoid
+    // to have multiple ongoing requests.
+    if (code === 'UND_ERR_BODY_TIMEOUT') {
```

Comment:
why?

Reply:
Use the modified test on main. The process hangs unrecoverable.